### PR TITLE
Cleanup environment variables

### PR DESCRIPTION
**Pull request by mvo5**
As discussed in the standup yesterday the current
 SNAP_APP_{,DATA,USER_DATA}_PATH
are too long. They are replaced with:
 SNAP_DIR, SNAP_DATA, SNAP_USER_DATA
in this branch. I also dropped SNAP_APP_TMPDIR because
we already set TMPDIR and TEMPDIR.

I'm open for alternative suggestions about the names of course.

### CODE REVIEW

**Comment by mvo5** [comment]
As discussed last week I renamed the variable from SNAP_DIR to SNAP. I can revert this change again easily of course. 

The upside of this change is that its shorter . The downside is that its potentially confisuing because not everything that has the SNAP_ prefix is a directory, we have: "SNAP_{NAME,VERSION,ORIGIN,FULLNAME,ARCH}" that is generic info and SNAP_{DATA,USER_DATA} that are directories.

Alternatives to "SNAP", "SNAP_DIR" might be: "SNAP_ROOT", "SNAP_BASE", "SNAP_HOME", "SNAP_PREFIX".

Feedback welcome!

**Comment by niemeyer** [comment]
Still LGTM after a weekend. :)

The suggested lack of a "type" in these variables is IMO okay. That's among the most common use cases for shell variables, the terseness seems worth it, and people already need some basic background information to be able to tell the distinction between, say, $SNAP_PATH and $SNAP_DATA_PATH.

Regarding $SNAP vs. $SNAP_BASE, I'd be okay with either (I'd avoid "root" and "home" terms to avoid the similarity with related concepts). I had a hard time finding such a term last week, but "base" does look okay. The reason why $SNAP alone doesn't bother me either is because I cannot imagine anything else we might use it for.. $SNAP would be a pointer the snap content itself, which is probably as close to "the snap" as we can get in a shell environment.

**Comment by elopio** [comment]
@sergiusens, @kyrofa: we'll have to update some of our examples.

**Comment by kyrofa** [comment]
@elopio yeah eventually-- and plugins. Thanks for deprecating though, @mvo5.

File: docs/security.md [hunk 1]
```diff
@@ -37,11 +37,10 @@ app launcher.
 Under the hood, the launcher:
 
 * sets up various environment variables (eg, `SNAP_APP_ARCH`,
```
Comment (chipaca) [inline docs/security.md]: oooh, while you're in here, could you fix this? it should be `SNAP_ARCH`.

File: helpers/helpers.go [hunk 2]
```diff
@@ -435,10 +434,15 @@ func GetUserSnapEnvVars(desc interface{}) []string {
 // somewhere more reasonable like the snappy module.
 func GetDeprecatedBasicSnapEnvVars(desc interface{}) []string {
 	return fillSnapEnvVars(desc, []string{
+		// SNAPP_
 		"SNAPP_APP_PATH={{.AppPath}}",
 		"SNAPP_APP_DATA_PATH=/var/lib{{.AppPath}}",
 		"SNAPP_APP_TMPDIR=/tmp/snaps/{{.UdevAppName}}/{{.Version}}/tmp",
 		"SNAPPY_APP_ARCH={{.AppArch}}",
```
Comment (chipaca) [inline helpers/helpers.go]: when are we killing the old deprecated ones? (rsn? please?)

Comment (chipaca) [inline helpers/helpers.go]: I say more: let's not ship 16.04 with any deprecated ones.

**Comment by chipaca** [comment]
:+1: :feelsgood:

File: docs/security.md (hunk 1 above)
Comment (mvo5) [inline docs/security.md]: Nice catch! Fixed.

File: helpers/helpers.go (hunk 2 above)
Comment (mvo5) [inline helpers/helpers.go]: I agree, lets kill SNAPP_ now and SNAP_ a bit later.

**Comment by mvo5** [comment]
I removed SNAPP_ now - those are *really* old. If I don't hear anything else I will merge after lunch.

**Pull request merged**